crates/turbopack-core/src/source_map/mod.rs

```diff
@@ -394,7 +394,11 @@ impl SourceMap {
                     let source = format!("/{SOURCE_MAP_ROOT_NAME}/{}", path_str);
                     let source_content = if let Some(source_content) = source_content {
                         source_content.to_string()
-                    } else if let FileContent::Content(file) = &*path.read().await? {
+                    } else if let Ok(FileContent::Content(file)) = path.read().await.as_deref() {
```

Comment:
I'm a bit concerned that this might hide some real file reading failure. I would prefer to convert an invalid path error into a `FileContent::NotFound` in the disk filesystem impl, so we don't miss real errors

Reply:
true, added a check for absolute paths to `turbo_tasks_fs::util::join_path`, let's see if that works